localstack/services/awslambda/invocation/runtime_executor.py

```diff
@@ -0,0 +1,233 @@
+import logging
+import re
+import shutil
+import time
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+from typing import Dict, Literal, Optional, Tuple
+
+from localstack import config
+from localstack.services.awslambda.invocation.executor_endpoint import (
+    ExecutorEndpoint,
+    ServiceEndpoint,
+)
+from localstack.services.awslambda.invocation.lambda_models import FunctionVersion
+from localstack.services.awslambda.lambda_utils import (
+    get_container_network_for_lambda,
+    get_main_endpoint_from_container,
+)
+from localstack.services.install import LAMBDA_RUNTIME_INIT_PATH
+from localstack.utils.archives import unzip
+from localstack.utils.container_utils.container_client import ContainerConfiguration
+from localstack.utils.docker_utils import DOCKER_CLIENT as CONTAINER_CLIENT
+from localstack.utils.net import get_free_tcp_port
+
+LOG = logging.getLogger(__name__)
+
+RUNTIME_REGEX = r"(?P<runtime>[a-z]+)(?P<version>\d+(\.\d+)?(\.al2)?)(?:.*)"
+
+# IMAGE_PREFIX = "gallery.ecr.aws/lambda/"
+IMAGE_PREFIX = "amazon/aws-lambda-"
+
+RAPID_ENTRYPOINT = "/var/rapid/init"
+
+InitializationType = Literal["on-demand", "provisioned-concurrency"]
+
+LAMBDA_DOCKERFILE = """FROM {base_img}
+COPY aws-lambda-rie {rapid_entrypoint}
+COPY code/ /var/task
+"""
+
+
+# TODO provided runtimes
+# TODO a tad hacky, might cause problems in the future.. just use mapping?
+def get_runtime_split(runtime: str) -> Tuple[str, str]:
+    match = re.match(RUNTIME_REGEX, runtime)
+    if match:
+        runtime, version = match.group("runtime"), match.group("version")
+        # sad exception for .net
+        if runtime == "dotnetcore":
+            runtime = "dotnet"
+            version = f"core{version}"
+        return runtime, version
+    raise Exception("Cannot process runtime '%s'" % runtime)
+
+
+def get_path_for_function(function_version: FunctionVersion) -> Path:
+    return Path(
+        f"{config.dirs.tmp}/lambda/{function_version.id.qualified_arn().replace(':', '_').replace('$', '_')}/"
+    )
+
+
+def get_code_path_for_function(function_version: FunctionVersion) -> Path:
+    return get_path_for_function(function_version) / "code"
+
+
+def get_image_name_for_function(function_version: FunctionVersion) -> str:
+    return f"localstack/lambda-{function_version.id.qualified_arn().replace(':', '_').replace('$', '_').lower()}"
+
+
+def get_image_for_runtime(runtime: str) -> str:
+    runtime, version = get_runtime_split(runtime)
+    return f"{IMAGE_PREFIX}{runtime}:{version}"
+
+
+def get_runtime_client_path() -> Path:
+    return Path(LAMBDA_RUNTIME_INIT_PATH)
+
+
+def prepare_image(target_path: Path, function_version: FunctionVersion) -> None:
+    if not function_version.config.runtime:
+        LOG.error("Images without runtime are currently not supported")
+        raise Exception("Custom images are currently not supported")
```

Comment:
nit: don't log and throw

maybe raise a `NotImplementedError` or `ValueError`?